src/vm/methodtablebuilder.cpp

```diff
@@ -5344,6 +5377,12 @@ MethodTableBuilder::PlaceVirtualMethods()
                                 it.Token());
                     }
                 }
+
+                // mazong1123: a valid overrided method.
+                if (!*hasOverridedEqualsMethod)
+                {
```

Comment:
Comparing method names and signatures is not free. Adding this kind of logic to type loader will make every type pay for it, but Equals will only ever be called on a few types. It would be better to do this lazily, only once somebody calls Equals on the type.